### ISSUE TITLE
Do not even parse excluded files

### ISSUE THREAD

**Issue by mvz** [troessner/reek#814]
Follow-up from #788:

> Note that this is different from excluding files in the Reek configuration: that will silence errors, but will not stop Reek from unsuccessfully trying to parse .rb files that are not valid Ruby code (such as .rb files that contain ERB instructions, as with Rails generator templates).

Needlessly parsing excluded files slows down analysis, so we should just skip them.

**Comment by troessner** [comment]
>> Note that this is different from excluding files in the Reek configuration: that will silence errors, but will not stop Reek from unsuccessfully trying to parse .rb files that are not valid Ruby code (such as .rb files that contain ERB instructions, as with Rails generator templates).

But that's not true. Excluding files via configuration will prevent them from being parsed:

>>
  ( $ ) cat config.reek
---
exclude_paths:
  - lib/legacy
  ( $ ) cat lib/legacy/unparseable.rb
def unpars%%%%ble; end
  ( $ ) cat lib/
legacy/ omg.rb
{-timo-|-columbia-2-} => [/tmp]
  ( $ ) cat lib/omg.rb
class Foo
end
  ( $ ) reek lib/
lib/omg.rb -- 1 warning:
  [1]:IrresponsibleModule: Foo has no descriptive comment [https://github.com/troessner/reek/blob/master/docs/Irresponsible-Module.md]

Without the config.reek I get:

>>
 ( $ ) reek lib/
lib/legacy/unparseable.rb:1:11: error: unexpected token tPERCENT
def unpars%%%%ble; end
          ^
lib/legacy/unparseable.rb: Parser::SyntaxError: unexpected token tPERCENT
lib/omg.rb -- 1 warning:
  [1]:IrresponsibleModule: Foo has no descriptive comment [https://github.com/troessner/reek/blob/master/docs/Irresponsible-Module.md]
1 total warning

And if you check the [source](https://github.com/troessner/reek/blob/master/lib/reek/source/source_locator.rb#L41) you can see that we skip those files. SourceLocator#sources is used by Input which is in turn used by OptionsInterpreter which is used by ReekCommand.

Long story short: `Reek` does not parse files that are under `exclude_paths`

**Comment by mvz** [comment]
Thanks for clearing that up, @troessner.
